[user] feat: 회원 소프트 삭제

LimdaeIl opened the pull request:
## #️⃣연관된 이슈
- 이슈 번호: #78 

## 변경 타입
- [x] 신규 기능 추가/수정
## 변경 내용

- **to-be**(변경 후 설명을 여기에 작성)

  - [x] TODO 회원 id 로 삭제

## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.
- [ ] 관련 테스트를 추가했습니다.
- [ ] 문서(코드, 주석, README 등)를 업데이트했습니다.

## 코멘트
- (추가적인 설명이나 코멘트가 필요한 경우 여기에 작성)

Commit:
[#90][user] feat: 회원 소프트 삭제

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/application/dto/DeleteUserRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+public record DeleteUserRequestDto(
+    Long deleterId) {
+
+  public static DeleteUserRequestDto of(DeleteUserRequestDto requestDto) {
```
자기 자신을 다시 반환하는 거 같은데 혹시 제가 잘못 이해한 걸까요?

LimdaeIl replied:
맞습니다... 이거 자기 자신을 반환하네요 .. 수정하겠습니다!

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/application/usecase/UserServiceImpl.java:
```diff
@@ -60,4 +63,12 @@ public PageResponse<GetAllUserResponseDto> getAllUsers(String username, String n
+  @Override
+  public DeleteUserResponseDto deleteUserByUserId(Long userId, DeleteUserRequestDto requestDto) {
+    User user = getUserByUserId(userId);
+    user.softDeleteUser(requestDto.deleterId(), LocalDateTime.now());
```
저희 BaseEntity 에 보면, delete() 라는 공통 메소드가 있습니다. 한 번 확인 부탁드립니다.

LimdaeIl replied:
넵! 다시 수정하겠습니다!

HanaHww2 commented on user/src/main/java/com/faster/user/app/user/application/usecase/UserServiceImpl.java:
```diff
@@ -60,4 +63,12 @@ public PageResponse<GetAllUserResponseDto> getAllUsers(String username, String n
+    User user = getUserByUserId(userId);
+    user.softDeleteUser(requestDto.deleterId(), LocalDateTime.now());
+
+    return DeleteUserResponseDto.of(userRepositoryAdapter.save(user).getId());
```
앗, 레포지토리를 인프라스트럭쳐에서 바로 가져오신 거 같네요. 다른 분들 코드 참고해서 도메인에 레포지토리 인터페이스 생성하시고, dip 로 구현해주실 수 있다면 그 방법이 좋을 거 같습니다.

LimdaeIl replied:
넵 다시 확인하고 수정하겠습니다!

HanaHww2 reviewed (approved):
수고하셨습니다. (●'◡'●)

Commit:
[#78][user] 회원 DIP 위배 수정 및 Response 응답 메서드명 수정

LimdaeIl commented:
@HanaHww2 @je-pa 다시 수정해서 커밋 올렸습니다!

HanaHww2 reviewed (commented):
수고하셨습니다!

Commit:
[#78][user] 회원 delete 메서드 사용 위치 수정 및 응답 DTO 사용 수정

of 를 처음에 잘못 사용해서 new 로 수정했다가, from 으로 올바르게 다시 수정
회원 엔티티 안에서 삭제 메서드 지우고 서비스에서 메서드 호출해서 사용하도록 수정

LimdaeIl commented:
@je-pa @HanaHww2 다시 수정해서 올려드렸습니다!

je-pa reviewed (approved):
고생하셨습니다!
